### PR TITLE
Add compatibility notes for using ActionText with ViewComponent

### PR DESCRIPTION
**Pull request by jaredplanter**
### Summary

Using ActionText with ViewComponent is not supported out of the box. It results in an error.

This updates the compatibility documents with a workaround until it is supported.

This has been previously reported in these issues:

https://github.com/github/view_component/issues/445
https://github.com/github/view_component/issues/351

Both proposed solutions. The latter being the simplest in requiring the least configuration.

Thank you @coderberry and @andrewmcodes: https://github.com/github/view_component/issues/351#issuecomment-631796883

### CODE REVIEW

File: docs/compatibility.md [hunk 1]
```diff
@@ -24,3 +24,18 @@ With the monkey patch disabled, use `render_component` (or  `render_component_to
...
+undefined method `rich_text_area_tag'
+```
+
+This is because ViewComponent does not have access to the helpers it needs via ActionText. To avoid this, add the following to your component (or base component):
+
+```
+delegate :rich_text_area_tag, to: :helpers
+```
+
+*_This is the simplest route. Adding the helpers directly via `include ActionText::Engine.helpers` leads to several more errors and requires additional configuration that is not necessary._
```
Comment (andrewmcodes) [inline docs/compatibility.md:41]: Looks great but I'm wondering if it's worth mentioning the other method without saying what it is or linking to it? Imo I believe this to be the "right" way to fix this and it doesn't just pertain to ActionText!

[Here's a great example](https://www.bridgetownrb.com/docs/components/ruby#helpers-1)

Comment (jaredplanter) [reply docs/compatibility.md]: Good call! I went ahead and removed it.

File: docs/compatibility.md [hunk 2]
```diff
@@ -24,3 +24,16 @@ With the monkey patch disabled, use `render_component` (or  `render_component_to
 ## Bridgetown (Static Site Generator)
 
 [Bridgetown](https://www.bridgetownrb.com/) supports ViewComponent via an experimental shim provided by the [bridgetown-view-component gem](https://github.com/bridgetownrb/bridgetown-view-component). More information available [here](https://www.bridgetownrb.com/docs/components/ruby#need-compatibility-with-rails-try-viewcomponent-experimental).
+
+## ActionText Support
```
Comment (joelhawksley) [inline docs/compatibility.md:28]:
```suggestion
## ActionText
```

File: docs/compatibility.md [hunk 3]
```diff
@@ -24,3 +24,16 @@ With the monkey patch disabled, use `render_component` (or  `render_component_to
...
+
+Trying to use `rich_text_area` via ActionText out of the box with ViewComponent will result in this error:
```
Comment (joelhawksley) [inline docs/compatibility.md:30]:
```suggestion
Using `rich_text_area` from ActionText in a ViewComponent will result in this error:
```

File: docs/compatibility.md [hunk 4]
```diff
@@ -24,3 +24,16 @@ With the monkey patch disabled, use `render_component` (or  `render_component_to
...
+```
+undefined method `rich_text_area_tag'
+```
```
Comment (joelhawksley) [inline docs/compatibility.md:33]:
```suggestion

```
undefined method `rich_text_area_tag'
```
```

File: docs/compatibility.md [hunk 5]
```diff
@@ -24,3 +24,16 @@ With the monkey patch disabled, use `render_component` (or  `render_component_to
...
+
+This is because ViewComponent does not have access to the helpers it needs via ActionText. To avoid this, add the following to your component (or base component):
```
Comment (joelhawksley) [inline docs/compatibility.md:35]:
```suggestion
This is due to ViewComponent not having access to the helpers it needs via ActionText. As a workaround, add the following to your component (or base component):
```

**Review by joelhawksley — APPROVED** [review]
👋🏻 mind adding a CHANGELOG?

File: docs/compatibility.md (hunk 4 above)
Comment (rmacklin) [reply docs/compatibility.md]: Hey @joelhawksley, I don't think this suggestion applied correctly due to using ``` inside the suggestion block

**Review by Spone — APPROVED** [review]

File: docs/CHANGELOG.md [hunk 6]
```diff
@@ -56,6 +56,10 @@ title: Changelog
 
     *Johannes Engl*
 
+* Add documentation for compatibility with ActionText
```
Comment (joelhawksley) [inline docs/CHANGELOG.md:59]:
```suggestion
* Add documentation for compatibility with ActionText.
```

**Pull request merged**